app/src/main/java/de/westnordost/streetcomplete/data/osmtracks/TracksApiImpl.kt

```diff
@@ -12,20 +12,24 @@ import de.westnordost.osmapi.traces.GpsTrackpoint
 import de.westnordost.streetcomplete.ApplicationConstants
 import de.westnordost.streetcomplete.data.download.ConnectionException
 import de.westnordost.streetcomplete.data.user.AuthorizationException
-import java.time.Instant
-import java.time.ZoneOffset
-import java.time.format.DateTimeFormatter
+import de.westnordost.streetcomplete.util.ktx.Instant
+import de.westnordost.streetcomplete.util.ktx.UTC
+import de.westnordost.streetcomplete.util.ktx.ZoneOffset
+import de.westnordost.streetcomplete.util.ktx.atZone
+import de.westnordost.streetcomplete.util.ktx.ofEpochMilli
+import kotlinx.datetime.toJavaInstant
 
 class TracksApiImpl(osm: OsmConnection) : TracksApi {
     private val api: GpsTracesApi = GpsTracesApi(osm)
 
     override fun create(trackpoints: List<Trackpoint>, noteText: String?): Long = wrapExceptions {
         // Filename is just the start of the track
         // https://stackoverflow.com/a/49862573/7718197
-        val name = DateTimeFormatter
-            .ofPattern("yyyy_MM_dd'T'HH_mm_ss.SSSSSS'Z'")
-            .withZone(ZoneOffset.UTC)
-            .format(Instant.ofEpochMilli(trackpoints[0].time)) + ".gpx"
+        val name = Instant.ofEpochMilli(trackpoints[0].time).atZone(ZoneOffset.UTC).let {
+            // doesn't seem to have a formatter function; could have done `.toJavaInstant()` and passed that to DateTimeFormatter but it would have been the 1 remaining instance of `java.time` --yoshi
+            ("${it.year.toPaddedString(4)}_${it.monthNumber.toPaddedString(2)}_${it.dayOfMonth.toPaddedString(2)}"
+                + "T${it.hour.toPaddedString(2)}_${it.minute.toPaddedString(2)}_${it.second.toPaddedString(2)}.${it.nanosecond.toPaddedString(6)}Z.gpx")
+        }
```

Comment:
Maybe create a private extension function to convert an Instant to a string in that format? I find this a bit hard to read.

Reply:
https://github.com/streetcomplete/StreetComplete/pull/4471/commits/077ce5add873e07888735c9ccf692e941967db79